Source/Timeline/AllDayView.swift

```diff
@@ -0,0 +1,163 @@
+     prevents conflicts trying to layout this view with the superview.width = 0
+     
+     More on this:
+     this scope of code is first invoked here:
+     
+     ````
+     @@ public class TimelineView: UIView, ReusableView, AllDayViewDataSource {
+     ...
+     public var layoutAttributes: [EventLayoutAttributes] {
+        ...
+        allDayView.reloadData()
+        ...
+     }
+     ````
+     
+     the superview.width is calcuated here:
+     
+     ````
+     @@ public class TimelineContainer: UIScrollView, ReusableView {
+     ...
+     override public func layoutSubviews() {
+        timeline.frame = CGRect(x: 0, y: 0, width: width, height: timeline.fullHeight)
+        ...
+     }
+     ````
+     */
+    svLeftConstraint.priority = UILayoutPriority(rawValue: 999)
+    
+    svLeftConstraint.isActive = true
+    sv.topAnchor.constraint(equalTo: topAnchor, constant: 2).isActive = true
+    sv.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+    bottomAnchor.constraint(equalTo: sv.bottomAnchor, constant: 2).isActive = true
+    
+    let maxAllDayViewHeight = allDayEventHeight * 2 + allDayEventHeight * 0.5
+    heightAnchor.constraint(lessThanOrEqualToConstant: maxAllDayViewHeight).isActive = true
+    
+    return sv
+  }()
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    
+    configure()
+  }
+  
+  required public init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+    
+    configure()
+  }
+  
+  // MARK: - RETURN VALUES
+  
+  // MARK: - METHODS
+  
+  private func configure() {
+    backgroundColor = UIColor.lightGray
+    clipsToBounds = true
+    reloadData()
+  }
+  
+  public func reloadData() {
+    guard let dataSource = self.dataSource else {
+      return
+    }
+    
+    defer {
+      layoutIfNeeded()
+    }
+    
+    // clear subviews TODO: clear out only contents of scroll view
+    scrollView.subviews.forEach { $0.removeFromSuperview() }
+    
+    let nEventDescriptors = dataSource.numberOfAllDayEvents(in: self)
+    if nEventDescriptors == 0 || nEventDescriptors < 0 { return }
+    
+    //TODO: add All-Day UILabel
+    
+    // create vertical stack view
+    let verticalStackView = UIStackView(frame: CGRect.zero)
+    verticalStackView.distribution = .fillEqually
+    verticalStackView.axis = .vertical
+    verticalStackView.spacing = 1.0
+    var horizontalStackView: UIStackView! = nil
+    
+    for index in 0...nEventDescriptors - 1 {
+      let eventDescriptor = dataSource.allDayView(self, eventDescriptorFor: index)
+      
+      // create event TODO: reuse event views
+      let eventView = EventView(frame: CGRect.zero)
+      eventView.updateWithDescriptor(event: eventDescriptor)
+      eventView.heightAnchor.constraint(equalToConstant: allDayEventHeight).isActive = true
+      
+      // create horz stack view if index % 2 == 0
+      if index % 2 == 0 {
+        horizontalStackView = UIStackView(frame: CGRect.zero)
```

Comment:
It might be a good idea to add this extension to the project:
```Swift
import UIKit

extension UIStackView {
  convenience init(axis: UILayoutConstraintAxis = .vertical,
                   distribution: UIStackViewDistribution = .fill,
                   alignment: UIStackViewAlignment = .fill,
                   spacing: CGFloat = 0,
                   subviews: [UIView] = []) {
    self.init(arrangedSubviews: subviews)
    self.axis = axis
    self.distribution = distribution
    self.alignment = alignment
    self.spacing = spacing
  }
}
```
As it makes code shorter and easier to read.

Reply:
👍 agreed, I haven't tried to refactor much, yet. But, this is a great first step.

Reply:
I use it in pretty much every project of mine, but that's more the question of a style.